code-challenges.test.js

```diff
@@ -1,46 +1,153 @@
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+    // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+    it("Return array of objects with names capped with sentence about person",()=> {
+      expect(capNamed2(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+    }
+  )})
+  //------Test failing with out function------
+  // FAIL  ./code-challenges.test.js
+  // capName
+  //   ✕ Return array of objects with names capped with sentence about person (3 ms)
+  // ● capName › Return array of objects with names capped with sentence about person
+  //   ReferenceError: capName is not defined
 
 // b) Create the function that makes the test pass.
+//-----Pseudo Code-------
+// Input: Array of Objects
+// Output: Array
+// Create a function called capName that has a parameter array.
+// (The first approach is messy and im sorry to either Gene or Nicole who has to read this) 
+// Well lets do this...you are going to map over the array that was passed into the function and the first portion of the function we are only worried about the key value pair <name>. Therefore, we need to use dot notation to acccess just the name in the object by using <value.name>. We then need to uppercase the first index of the value.name and then slice everything else after that and add it to the capitilizaed first letter. So we are slicing from index[1] to the first emepty space in the name + 1 and then adding that sliced portion back to the first letter we capitilized. (that only takes care of the first name, we still have the last name to cap). To cap the last name we need to find the character after the first empty space and cap it we do this by finding the indexOf(" ") +1 of value.name and use the built in  method toUpperCase (value.name.charAt(value.name.indexOf(" ") + 1).toUpperCase()). Great so far we have the fisrt name capped and concated back together with the first letter of the last name capped. Now we just need to concat the rest of the last name back together. we do this by slicing from the empty space +2 so we ignore the fisrt letter because we already have that, by  indexOf(" ") +2. and we concat everything after that point to the first letter of the last name that we capped. We now have the first and last name capped.
+// now we just need to concat the occupation to our newly capped first and last name
+// then return this disaster of concatenation 
+
+const capName = (array) => {
+  return array.map((value) => {
+   return value.name[0].toUpperCase() + value.name.slice(1, value.name.indexOf(" ") + 1) + value.name.charAt(value.name.indexOf(" ") + 1).toUpperCase() + value.name.slice(value.name.indexOf(" ") + 2) + " is " + 
+   value.occupation + "."
+  })
+}
+
+//----Refactored because the first one is UGLY-----
```

Comment:
Thanks for leaving it in here! Its really cool to see the different approach and how you refactored! 💯